src/composables/trade/useTradeState.ts

```diff
@@ -0,0 +1,35 @@
+import { reactive, toRefs } from 'vue';
+
+// globals
+const tradeState = reactive({
+  tokenInAddress: '',
+  tokenOutAddress: '',
+  tokenInAmount: '',
+  tokenOutAmount: ''
+});
+
+export function useTradeState() {
+  function setTokenInAddress(address: string) {
+    tradeState.tokenInAddress = address;
+  }
+  function setTokenOutAddress(address: string) {
+    tradeState.tokenOutAddress = address;
+  }
+
+  function setTokenInAmount(amount: string) {
+    tradeState.tokenInAmount = amount;
+  }
+  function setTokenOutAmount(amount: string) {
+    tradeState.tokenOutAmount = amount;
+  }
```

Comment:
I think it would be nice to move these functions outside the composable (but still return them) so that they are defined only once.

Reply:
good point